### PR TITLE
Start basic Git styleguide

### PR DESCRIPTION
**Pull request by duggiefresh**
This is not intended to be the full encompassing guide on Git, and only
a basic guide. I'd like to provide basic suggestions to the team, in order
to make our commits more consistent. Project teams and individuals can use
this as a guide, and are free to make modifications as they see fit.

Signed-off-by: Doug Yun <doug.yun@dockyard.com>

### CODE REVIEW

File: general/git.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+The message body of the commit is *optional*, however, it is beneficial to take
+the time to convey the *why* and *how* in the commit body.
+
+Let's take a look at Tim Pope's [example of a model Git commit message.][pope-git-commit]
+
+```bash
+Capitalized, short (50 chars or less) summary
+
+More detailed explanatory text, if necessary.  Wrap it to about 72
```
Comment (aadam21) [inline general/git.md:117]: Yes!

File: general/git.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+When commit messages answer questions like these, our audience - colleagues and clients -
+will have a better understanding in time of what our intention was.
+
+* Descriptive commit title
+* Include supporting ticket number
+* What and Why
+
+How do we inform our audience?
+
+Let's take a look at some examples of commits for an Emac's init file.
```
Comment (aadam21) [inline general/git.md:25]: Wouldn't this be `Emacs`?

File: general/git.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+```
+
+Now using the set of well written commits, let's try searching for the keyword `dired`
+again.
+
+```bash
+
+# Ideal
+
+$ git log --oneline -1
```
Comment (aadam21) [inline general/git.md:92]: Do you think there would be any value lost in removing this and illustrating the "happy path" with the `--grep` used below to eliminate some duplication?

Comment (duggiefresh) [reply general/git.md:92]: Good point! I only included the `git log` portion to show that there was a commit title that was descriptive with important keywords. If you find it to be duplicative work, I can remove the `git log` and highlight that the descriptive title made the `grep` possible.

Comment (aadam21) [reply general/git.md:92]: I don't honestly have a strong opinion either way. I just noticed on first reading I had to do a double take to make sure I didn't re-read or misread a line.

File: general/git.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+1. [Commit Messages](#commit-messages)
+   1. [Writing Message Titles](#writing-message-titles)
+   1. [Writing Message Bodies](#writing-message-bodies)
+
+## Commit Messages
+
+Commit messages are an opportunity for us to convey a brief story or synopsis of
+work completed. We should be answering questions like:
+
+*What caused this original regression and why does this newly introduced code fix it?*,
```
Comment (monicawheeler) [inline general/git.md:14]: _What do you think of changing the format, and some minor grammar updates, to this?_

Commit messages are an opportunity for us to convey a brief story or synopsis of
work completed. We should be answering questions like:

- What caused this original regression and why does this newly introduced code fix it?
- What dependency was removed from the application, and how does this affect performance?

When commit messages answer these questions, our audience, colleagues and clients, will have a better understanding of what our intention was.

File: general/git.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+*What dependency was removed from the application, and how does this affect performance?*
+When commit messages answer questions like these, our audience - colleagues and clients -
+will have a better understanding in time of what our intention was.
+
+* Descriptive commit title
```
Comment (monicawheeler) [inline general/git.md:19]: _It might be helpful to add what these bulleted items represent. For example:_

The following is a list of items you should include in your commit message: 

* Descriptive commit title
...
...

File: general/git.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+* Include supporting ticket number
+* What and Why
+
+How do we inform our audience?
+
+Let's take a look at some examples of commits for an Emac's init file.
+
+### Writing Message Titles
+
+Commit message titles should be descriptive and succinct. Imagine that you
```
Comment (monicawheeler) [inline general/git.md:29]: Recommended change:

```
Commit message titles should be descriptive and succinct. Imagine that you
have to scan a list of commit titles and messages. What characteristics would
make that task easier?
```

File: general/git.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+have to scan a list of commit titles and messages - what characteristics would
+make that task easier?
+
+Let's take a look at a set of commit titles that are inconsistent with varying
+degrees of information.
+
+```bash
+
+# Not ideal
```
Comment (monicawheeler) [inline general/git.md:38]: Love having the examples showing ideal and not ideal solutions! What do you think about pulling the `# Not ideal` and `# Ideal` out of the code snippets and into a subheader to clearly label the types of code snippets?

File: general/git.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+$ git log --oneline -1 --grep='dired'
+
+7411bd8 Add `auto-revert` to `dired` mode
+# We see the desired commit!
+
+```
+
+### Writing Message Bodies
+
+Just like descriptive commit titles, the content of the commit can provide
```
Comment (monicawheeler) [inline general/git.md:106]: 💯 Love this!

**Comment by duggiefresh** [comment]
Thanks all for commenting! I've updated the PR with a section about "Committing with Multiple Authors", and fixed grammar and sentence structure.

File: general/git.md (hunk 7 above)
Comment (duggiefresh) [reply general/git.md]: 👍 I've pulled out the "not ideal" and "ideal" into subheaders. I also went with "not recommended" and "recommended." Any thoughts on the word update?

File: general/git.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+
+Commit message titles should be descriptive and succinct. Imagine that you
+have to scan a list of commit titles and messages. What characteristics would
+make that task easier?
+
+Let's take a look at some examples of commits for an Emacs `init` file.
+Here is a set of commit titles that are inconsistent with varying
+degrees of information.
+
+#### Not Recommended
```
Comment (monicawheeler) [inline general/git.md:37]: 👍🏻

File: general/git.md [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+reference the ticket number. Suggested areas are the top or bottom of the commit
+message body. Thanks all!
+
+Close #1234
+
+```
+
+[pope-git-commit]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
+
+### Committing with Multiple Authors
```
Comment (monicawheeler) [inline general/git.md:163]: 👍🏻

**Pull request merged**